apps/portal/app/buckets/[address]/_components/metadata-panel.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * A component to display object metadata in a more compact format.
+ * Designed to be used alongside the file preview area.
+ */
+import { ChevronDown, ChevronUp } from "lucide-react";
+import { useState } from "react";
+
+import CollapsedStringDisplay from "@recallnet/ui/recall/collapsed-string-display";
+
+import Metric from "@/components/metric";
+import { arrayToDisplay } from "@/lib/convert-matadata";
+
+interface MetadataPanelProps {
+  object: {
+    blobHash: string;
+    recoveryHash: string;
+    size: bigint;
+    expiry: bigint;
+    metadata: readonly {
+      key: string;
+      value: string;
+    }[];
+  };
+  objectSize: {
+    val: string | number;
+    unit: string | undefined;
+    formatted?: string;
+  };
+  objectExpiryDisplay?: string;
+  objectExpiryIso?: string;
+  objectBlockDiff?: bigint;
+}
+
+export function MetadataPanel({
+  object,
+  objectSize,
+  objectExpiryDisplay,
+  objectExpiryIso,
+  objectBlockDiff,
+}: MetadataPanelProps) {
+  const [expanded, setExpanded] = useState(false);
+
+  // Convert number val to string if needed
+  const sizeVal =
+    typeof objectSize.val === "number"
+      ? objectSize.val.toString()
+      : objectSize.val;
```

Comment:
If you make the changes suggested above, you can remove this and access `objectSize.val` directly.